plugins/modules/license_keys.py

```diff
@@ -0,0 +1,291 @@
+from ..module_utils.sap_id_sso import sap_sso_login
+
+
+DOCUMENTATION = r'''
+---
+module: license_keys
+
+short_description: Creates systems and license keys on me.sap.com/licensekey
+
+version_added: 1.1.0
+
+options:
+  suser_id:
+    description:
+      - SAP S-User ID.
+    required: true
+    type: str
+  suser_password:
+    description:
+      - SAP S-User Password.
+    required: true
+    type: str
+  installation_nr:
+    description: 
+      - Number of the Installation for which the system should be created/updated
+    required: true
+    type: str
+  system:
+    description:
+      - The system to create/update
+    required: true
+    type: dict
+    suboptions:
+      nr:
+        description:
+          - The number of the system to update. If this attribute is not provided, a new system is created.
+        required: false
+        type: str
+      product:
+        description:
+          - The product description as found in the SAP portal, e.g. SAP S/4HANA
+        required: true
+        type: str
+      version:
+        description:
+          - The description of the product version, as found in the SAP portal, e.g. SAP S/4HANA 2022
+        required: true
+        type: str
+      data:
+        description: 
+          - The data attributes of the system. The possible attributes are defined by product and version.
+          - Running the module without any data attributes will return in the error message which attributes are supported/required.
+        required: true
+        type: dict
+
+  licenses:
+    description:
+      - List of licenses to create for the system.
+      - If the license does not exist, it is created.
+      - If it exists, it is updated.
+    required: true
+    type: list
+    elements: dict
+    suboptions:
+      type:
+        description:
+          - The license type description as found in the SAP portal, e.g. Maintenance Entitlement
+        required: true
+        type: str
+      data:
+        description: 
+          - The data attributes of the licenses. The possible attributes are defined by product and version.
+          - Running the module without any data attributes will return in the error message which attributes are supported/required
+          - In practice, most license types require at least a hardware key (hwkey) and expiry date (expdate)
+        required: true
+        type: dict
+      
+  delete_other_licenses:
+    description:
+      - Whether licenses other than the ones specified in the licenses attributes should be deleted.
+      - This is handy to clean up older licenses automatically.
+    type: bool
+    required: false
+    default: false
+    
+  
+author:
+    - Lab for SAP Solutions
+
+'''
+
+
+EXAMPLES = r'''
+- name: create license keys
+  community.sap_launchpad.license_keys:
+    suser_id: 'SXXXXXXXX'
+    suser_password: 'password'
+    installation_nr: 12345678
+    system:
+      nr: 12345678
```

Comment:
is installation_nr and nr always the same ? If yes why define it twice

Reply:
installation_nr is the installation number (usually you have 1 installation for your company which contains all S4HANAs, 1 for all BW4HANAs etc) and system.nr is the system number, i.e. one specific installation of S4HANA, for example your QA, Test or Prod system.

Reply:
re-reading my sentence just know makes it obvious why it's confusing ;) installation is a very confusing term. however, that's what SAP chose, so I'd say we use it too.

Reply:
I changed the sample values so that it's more obvious that those are different values